contracts/dtoken/src/supply.rs

```diff
@@ -77,13 +79,27 @@ impl Contract {
             U128(self.total_reserves),
             U128(interest_rate_model.get_reserve_factor()),
         );
+        let accrued_interest = self.get_accrued_supply_interest(env::signer_account_id());
         let accrued_supply_interest = interest_rate_model.calculate_accrued_interest(
             supply_rate,
             self.get_supplies_by_account(env::signer_account_id()),
-            self.get_accrued_supply_interest(env::signer_account_id()),
+            accrued_interest.clone(),
         );
         self.set_accrued_supply_interest(env::signer_account_id(), accrued_supply_interest);
 
+        for reward_setting in self.model.rewards_config.clone().iter() {
+            let reward_amount = reward_setting.reward_per_day /* * user.stake / self.token.total_supply * ((block_height() - accrued_interest.last_recalculation_block) as u128/ blocks_per_day)*/;
```

Comment:
Let's change parameter reward_per_day to
reward_amount: RewardsAmount

RewardsAmount {
  period: Enum::Day || Enum::Week,
  amount: WBalance,
}